jpeg_benchmark.py

```diff
@@ -0,0 +1,175 @@
+REAL_FPS = 30
+FRAME_TO_SAVE = 69
+INPUT_PATH = pathlib.Path("test_images", "Encode Test Dataset 2024")
+OUTPUT_PATH = pathlib.Path(f"log_{int(time.time())}")
+ENCODING = "jpeg"
+NUM_REPEATS = 1
+
+# Keys for dictionary entries
+TOTAL_TIME_MS = "total_time_ms"
+AVG_TIME_MS = "avg_time_ms"
+TOTAL_SPACE_B = "total_space_B"
+AVG_SPACE_B = "avg_space_B"
+FRAME_SPACE = "frame_space"
+FRAME_DATA = "frame_data"
+
+STIMULATED_FPS = [1, 5, 10, 30] # Test image compression, FPS
+QUALITY_SETTINGS = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
+
+# For output csv file
+HEADERS = [
+    "Quality",
+    "Stimulated FPS",
+    "Total Time (ms)",  
+    "Avg Time (ms)",
+    "Max Space (B)",
+    "Avg Space (B)",
+    "Frame Space",
+]
+HEADER_LINE = ",".join(HEADERS) + "\n"
+
+def update_min_max(min_value: "int | float",
+                   max_value: "int | float",
+                   current_value: "int | float",) -> "tuple[int, int] | tuple[float, float]":
+    """
+    Updates the min and max values for a measurement.
+
+    Args:
+        min_value: previous minimum value
+        max_value: previous maximum value
+        current_value: currently measured value
+
+    Returns: (min_value, max_value)
+        min_value: new updated minimum recorded value
+        max_value: new updated maximum recorded value
+        
+        The intended output is something like [int, int] or [float, float],
+        but it is not guaranteed because the inputs could be a combination of int and float.
+        eg. could also be tuple[float, int]
+    """
+    if current_value < min_value:
+        min_value = current_value
+    if current_value > max_value:
+        max_value = current_value
+
+    return min_value, max_value
+
+def run():
+    OUTPUT_PATH.mkdir(parents=True, exist_ok=True)
+    print("\n--------------Loading Images----------------\n")
+    images = [Image.open(INPUT_PATH / f"{i}.png") for i in range(FRAME_COUNT)]
+    print("\n--------------Images Loaded----------------\n")
+
+    # Set up results dictionary
+    results = {
+        f"lossy_{quality}": {
+            f"stimulated_fps_{sfps}": {
+                TOTAL_TIME_MS: 0,
+                AVG_TIME_MS: 0,
+                TOTAL_SPACE_B: 0,
+                AVG_SPACE_B: 0,
+                FRAME_SPACE: int(FRAME_COUNT/REAL_FPS * sfps),
+                FRAME_DATA: [],
+            } for sfps in STIMULATED_FPS
+        } for quality in QUALITY_SETTINGS
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+    for quality in QUALITY_SETTINGS:
+        quality_directory = "lossless" if quality == 100 else f"loss_{quality}"
+        if not os.path.exists(f"./results/{quality_directory}"):
+            os.mkdir(f"./results/{quality_directory}")
+
+        for sfps in STIMULATED_FPS:
+            TOTAL_SPACE_B = 0
+            if not os.path.exists(f"./results/{quality_directory}/{sfps}fps"):
+                os.mkdir(f"./results/{quality_directory}/{sfps}fps")
```

Comment:
No need for all of these folders if only saving 1 sample image per test. (see other comments)